app/models/compound_aliquot.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+class CompoundAliquot # rubocop:todo Style/Documentation
+  include ActiveModel::Model
+
+  DUPLICATE_TAG_DEPTH_ERROR_MSG = "Cannot create compound sample from following samples due to duplicate 'tag depth'"
+  MULTIPLE_STUDIES_ERROR_MSG =
+    'Cannot create compound sample due to the component samples being under different studies.'
+
+  attr_accessor :request, :source_aliquots
+
+  attr_reader :component_samples
+
+  validate :tag_depth_is_unique
+
+  def initialize(attributes)
+    super
+
+    @component_samples = source_aliquots.map(&:sample)
+  end
+
+  # Check that the component samples in the compound sample will be able to be distinguished -
+  # this is represented by them all having a unique 'tag_depth'
+  def tag_depth_is_unique
+    return unless source_aliquots.pluck(:tag_depth).uniq.count != source_aliquots.size
+
+    errors.add("#{DUPLICATE_TAG_DEPTH_ERROR_MSG}: #{component_samples.map(&:name)}")
+  end
+
+  def default_library_type
+    source_aliquots.first.library_type
+  end
+
+  # Default study that the new compound sample will use
+  # Uses the one from the request if it's present,
+  # otherwise, the one from the source aliquots if it's consistent.
+  def default_compound_study
+    request.initial_study ||
+      begin
+        raise MULTIPLE_STUDIES_ERROR_MSG if studies.count > 1
```

Comment:
Possibly move this to validation instead, so the check can be performed up-front before you do anything.

Reply:
Yeah I wondered whether to, but the reason it's here is because you only really care about this check if there's no 'initial_study'. I suppose we could also do that check in the validation though.

Reply:
You can make validation conditional: `validate :aliquot_studies_unique, unless: :initial_study` rather than including it in one method